packages/flutter/test/cupertino/nav_bar_transition_test.dart

```diff
@@ -1020,6 +1020,76 @@ void main() {
     );
   });
 
+  testWidgets('Bottom CupertinoSliverNavigationBar.bottom fades and slides in from the left', (
+    WidgetTester tester,
+  ) async {
+    await startTransitionBetween(
+      tester,
+      from: const CupertinoSliverNavigationBar(
+        bottom: PreferredSize(preferredSize: Size.fromHeight(30.0), child: Placeholder()),
+      ),
+      fromTitle: 'Page 1',
+    );
+
+    await tester.pump(const Duration(milliseconds: 50));
+
+    // There's 2, one from the bottom large title fading out and one from the
+    // bottom back label fading in.
+    expect(flying(tester, find.text('Page 1')), findsNWidgets(2));
+    expect(flying(tester, find.byType(Placeholder)), findsOneWidget);
+
+    checkOpacity(tester, flying(tester, find.byType(Placeholder)), 0.996379017829895);
+
+    expect(
+      tester.getTopLeft(flying(tester, find.byType(Placeholder))),
+      const Offset(-20.579326152801514, 96.0),
```

Comment:
Instead of these precise numbers I'd prefer golden tests for easier maintenance.

Reply:
We tend to do tests like these for page transitions, instead of goldens. Especially for the Cupertino slide transitions. I think golden tests for transitions tend to break for unrelated changes and can be difficult to debug if the changes are expected or not.

We could add a range to make them a little more robust against failures though, [like other tests](https://github.com/flutter/flutter/blob/6b5b468a601704d8b19db90dccd64634c05532ba/packages/flutter/test/cupertino/route_test.dart#L747).